Allow LOO of samples from a downsampled beagle

eriqande opened the pull request:
Hey Matt!

Nothing to do right now.  I'm just using this pull request to let you know I am implementing a few new features in WGSassign.  I am working this summer with Nina (@nt246) on some fisheries applications of WGSassign.  I wanted to do some downsampling tests like we did in the original MIEE paper, but we didn't have a lot of test samples available.  However, I realized that we could work the downsampling into the LOO procedure pretty darn easily.  

I am done implementing the --loo_downsampled_beagle option.  See the new section in the README:
**Leave-one-out cross validation and test with downsampled GLs for samples being assigned**
for details.

I am going to keep developing some new things on this branch---most notably an internal implementation of checking consistency of assignments on partitioned data sets.  It should be super straightforward to just break the sites into different subsets and calculate the log-likelihoods for each, all within WGSassign.  

I'll be using these modifications over the next month or so, so they should get pretty well tested.   I will let you know when I think you should fully pull them in.  

I hope you are doing great and having a good summer.

eriqande committed:
Allow LOO of samples from a downsampled beagle

See the new section in the README:

**Leave-one-out cross validation and test with downsampled GLs for samples being assigned**

for details.

mgdesaix commented:
awesome, sounds good. Can you believe we archived the last Zenodo version over 2 years ago?? Just saw that, wow time flies!

This is great. And thanks for looking into implementing an internal assignment consistency...I'd written some bash scripts for ConGen to facilitate that last year but never got around to putting that functionality into WGSassign. Looking forward to hearing what you and Nina are up to with this! Talk soon

eriqande committed:
Partway done with in-program site-partitioning

Just committing and pushing this to back up the work.  Significant changes:

- `--partition_sites` INT option says to break up sites int INT groups defined by the remainder after dividing each site's index by INT.
- Major possibly breaking change for some users' scripts: Write logls out not to a matrix with no sample names or column names.  Now it uses pandas to insert that information and write it out.  For LOO, it includes the actual source population.  For partitioned sites, includes a column of which data partition (data_part) it is.

Still have to do the same for assignLL

eriqande commented:
Great!  I just pushed up a partial commit on the internal partitioning for consistency checks.  More soon.

eriqande committed:
gitignore silly Mac files

eriqande committed:
Fix critical bug! Proper downsampling LOO now!

I had forgotten to add the `downsampled_L=L_ds` argument to glassy.loo for the downsampling part.  Got that taken care of now.
